### PR TITLE
Fonts and chipping away at more base styles.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW 

# Summary
- Form elements and fonts

# Needed By (Date)
- Whenever

# Steps to Test

1. Copy sample.test.html into development environment
2. Start the development server and review the form elements
3. Ensure the fonts are loading.

### CODE REVIEW

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

File: src/css/fonts.css [hunk 1]
```diff
@@ -0,0 +1,102 @@
+@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,200i,300,300i,400,400i,600,600i,700,700i,900,900i");
```
Comment (yvonnetangsu) [inline src/css/fonts.css:1]: I'm thinking that we should remove the 200 font weight since even 300 isn't accessible for some font sizes (and super thin on some browsers).

File: src/css/fonts.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+@import url("https://fonts.googleapis.com/css?family=Source+Serif+Pro:400,600,700");
+@import url("https://fonts.googleapis.com/css?family=Roboto+Slab:300,400,700");
+@import url("https://fonts.googleapis.com/css?family=Kalam:300,400,700");
```
Comment (yvonnetangsu) [inline src/css/fonts.css:4]: Kalam and Noto-sans might go but that probably need a confirmation :)

File: src/css/fonts.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  src: url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.eot");
+  src: url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.eot?#iefix") format("embedded-opentype"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.woff2") format("woff2"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.woff") format("woff"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.ttf") format("truetype"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-brands-400.svg#fontawesome") format("svg");
+}
+
+/** Import Stanford Hosted Fonts **/
+@font-face {
```
Comment (yvonnetangsu) [inline src/css/fonts.css:35]: Good news is that these Source Serif Pro variants is now available on Google Fonts so we can just add that to the top Google fonts import line - they might not have the font weight 200 for this family (and we probably shouldn't use for a11y reasons too).

**Comment by sherakama** [comment]
I'm not 100% on this approach yet either. 

I wanted to bundle the fonts in a plugin but it looks like I'd have to add another unofficial plugin to be able to do that so I punted and added the fonts.css file.

Also, this bloats and complicates the package with all of the font assets so it might be nice to break this off into its own repo/package. eg: `decanter/fonts` for anything that isn't at a remote url.

**Comment by yvonnetangsu** [comment]
> I'm not 100% on this approach yet either.
> 
> I wanted to bundle the fonts in a plugin but it looks like I'd have to add another unofficial plugin to be able to do that so I punted and added the fonts.css file.
> 
> Also, this bloats and complicates the package with all of the font assets so it might be nice to break this off into its own repo/package. eg: `decanter/fonts` for anything that isn't at a remote url.

Yeah, I figured you're still working on this. We got so many fonts..... and more new ones coming with the new Identity Guide 🤔

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

**Comment by sherakama** [comment]
It looks like we will be needing to keep fonts.css for the time being.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

File: src/css/fonts.css [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,600,600i,700,700i,900,900i");
+@import url("https://fonts.googleapis.com/css?family=Source+Serif+Pro:400,600,700");
+@import url("https://fonts.googleapis.com/css?family=Roboto+Slab:300,400,700");
+@import url("https://fonts.googleapis.com/css?family=Kalam:300,400,700");
+@import url("https://fonts.googleapis.com/css?family=Noto+Sans:400,400i,700,700i&subset=devanagari");
+@import url("https://fonts.googleapis.com/css2?family=Source+Serif+Pro:ital,wght@0,300;0,400;0,600;0,700;0,900;1,200;1,300;1,400;1,600;1,700;1,900&display=swap");
+
+/** Import Font Awesome Fonts */
+@font-face {
+  font-family: 'Font Awesome 5 Free';
```
Comment (yvonnetangsu) [inline src/css/fonts.css:12]: I have a feeling that we're going to need the FA solids also, now that they're used in the new alerts, but that could be in another PR.

File: src/css/fonts.css [hunk 5]
```diff
@@ -0,0 +1,41 @@
+@charset "UTF-8";
+
+@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,600,600i,700,700i,900,900i");
+@import url("https://fonts.googleapis.com/css?family=Source+Serif+Pro:400,600,700");
+@import url("https://fonts.googleapis.com/css?family=Roboto+Slab:300,400,700");
+@import url("https://fonts.googleapis.com/css?family=Kalam:300,400,700");
```
Comment (yvonnetangsu) [inline src/css/fonts.css:6]: Are we still supporting Kalam and Noto Sans for v7?

File: src/css/fonts.css [hunk 6]
```diff
@@ -0,0 +1,41 @@
+@charset "UTF-8";
+
+@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,600,600i,700,700i,900,900i");
+@import url("https://fonts.googleapis.com/css?family=Source+Serif+Pro:400,600,700");
```
Comment (yvonnetangsu) [inline src/css/fonts.css:4]: Do we still need this Source Serif Pro line now that we have the one below at line 8? Looks like that one includes both regular and italic.

File: src/plugins/base/base.js [hunk 7]
```diff
@@ -204,8 +204,154 @@ module.exports = function () {
...
+        borderRadius: '2rem',
+      },
+      '[type="range"]:focus::-webkit-slider-thumb': {
+        border: '2px solid ' + config('theme.colors.fog.dark'),
+      },
+      '[type="range"]:focus::-moz-range-thumb': {
+        border: '2px solid ' + config('theme.colors.fog.dark'),
+      },
+      '[type="range"]:focus::-ms-thumb': {
+        border: '2px solid ' + config('theme.colors.fog.dark'),
```
Comment (yvonnetangsu) [inline src/plugins/base/base.js:353]: I have a feeling we'll work on these some more anyway, so I'm not going to do a detailed review of these. They look fine in the Tugboat view for me.

File: src/plugins/utilities/layout/centered-container.js [hunk 8]
```diff
@@ -18,7 +18,7 @@ module.exports = function () {
 
     const newUtilities = {
       // Center an element horizontally.
-      '.centered-container': {
+      '.centered-container, .cc': {
```
Comment (yvonnetangsu) [inline src/plugins/utilities/layout/centered-container.js:21]: I can see shortening some class name can be useful since there'll be so many.

**Review by yvonnetangsu — COMMENTED** [review]
Just a few font related questions, and I can't seem to do `npm run watch` in /dev anymore - was looking at things on Tugboat. I saw that you commented out the dev server stuff in webpack.dev.config.js so that's probably expected? Otherwise looking good - thanks for working on this!

File: src/css/fonts.css (hunk 4 above)
Comment (sherakama) [reply src/css/fonts.css:12]: Good point. Let's pull those in.

File: src/css/fonts.css (hunk 5 above)
Comment (sherakama) [reply src/css/fonts.css:6]: I'm not sure, I would like to keep a 1:1 with v6 where possible but if we are going to remove them, we should make a note as deprecated in v6.

Comment (sherakama) [reply src/css/fonts.css:6]: Who gets to make that call? @kerri-augenstein perhaps?

File: src/css/fonts.css (hunk 6 above)
Comment (sherakama) [reply src/css/fonts.css:4]: Guess not.
```suggestion
```

File: src/plugins/utilities/layout/centered-container.js (hunk 8 above)
Comment (sherakama) [reply src/plugins/utilities/layout/centered-container.js:21]: I find we use this one a lot so it made sense to me.

File: src/plugins/base/base.js (hunk 7 above)
Comment (sherakama) [reply src/plugins/base/base.js:353]: Yeah, starting point or base style at the very least.

From here we will create our form components which will be much more involved.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

File: src/css/fonts.css (hunk 5 above)
Comment (yvonnetangsu) [reply src/css/fonts.css:5]: Yeah @kerri-augenstein and Dana at UComm both agreed to it in a previous discussion, but let's confirm with them again before making any decisions. Please feel free to leave them in for this PR.

Comment (sherakama) [reply src/css/fonts.css]: > Kerri Augenstein  @Shea we haven't needed script at all, so no on Kalam, and I have no recollection of the Noto Sans and where that came from. I'd say no on that as well.

**Comment by sherakama** [comment]
@yvonnetangsu Ready to go again

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

File: src/css/fonts.css [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  src: url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.eot");
+  src: url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.eot?#iefix") format("embedded-opentype"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.woff2") format("woff2"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.woff") format("woff"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.ttf") format("truetype"),
+  url("~@fortawesome/fontawesome-free/webfonts/fa-regular-400.svg#fontawesome") format("svg");
+}
+
+@font-face {
+  font-family: 'Font Awesome 5 Solid';
```
Comment (yvonnetangsu) [inline src/css/fonts.css:22]: Here's the weird thing. Both solids and regulars actually have the same font-family name "Font Awesome 5 Free" (like what you did for v6 here)
https://github.com/SU-SWS/decanter/blob/master/core/src/scss/core/fonts/_fontawesome-fonts.scss

File: src/plugins/theme/fontFamily.js [hunk 10]
```diff
@@ -10,5 +10,6 @@ module.exports = function () {
     stanford: ['Stanford','"Source Serif Pro"', 'Georgia', 'Times','"Times New Roman"', 'serif'],
     fab: ['"Font Awesome 5 Brands"'],
     far: ['"Font Awesome 5 Free"'],
+    fas: ['"Font Awesome 5 Solid"'],
```
Comment (yvonnetangsu) [inline src/plugins/theme/fontFamily.js:13]: See above comment

**Review by yvonnetangsu — COMMENTED** [review]
Just that one thing - other than that ready to merge :)

File: src/plugins/theme/fontFamily.js (hunk 10 above)
Comment (sherakama) [reply src/plugins/theme/fontFamily.js:13]: Huh, yeah, but with different class names... Now why did they do that...

https://github.com/FortAwesome/Font-Awesome/blob/master/scss/solid.scss
https://github.com/FortAwesome/Font-Awesome/blob/master/scss/regular.scss

Comment (yvonnetangsu) [reply src/plugins/theme/fontFamily.js:13]: I know....crazy right? 😂

Comment (sherakama) [reply src/plugins/theme/fontFamily.js:13]: Ah, cuz the names of the icons are different. hmm. kay.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr723-kt25hwxfqbz2iwncpfzorc7hra5mm4o3.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f44a9066bc75ae69f02724a

**Review by yvonnetangsu — APPROVED** [review]
GTG woohoo! 🎉

**Comment by sherakama** [comment]
@yvonnetangsu 

> and I can't seem to do npm run watch in /dev anymore

You should run `npm run watch` & `npm run start` from the root of the repo and it should work. I didn't comment anything out in this PR.
